frontend/src/metabase/admin/permissions/selectors/data-permissions/permission-editor.tsx

```diff
@@ -400,12 +400,16 @@ export const getGroupsDataPermissionEditor: GetGroupsDataPermissionEditorSelecto
 
       const entities = sortedGroups.map((group) => {
         const isAdmin = isAdminGroup(group);
-        const isExternal =
-          !!externalUsersGroup && PLUGIN_TENANTS.isExternalUsersGroup(group);
+
+        const isAllTenantUsersGroup =
+          !!allTenantUsersGroup && PLUGIN_TENANTS.isExternalUsersGroup(group);
 
         const isTenantGroup = PLUGIN_TENANTS.isTenantGroup(group);
         let groupPermissions;
 
+        const shouldUseAllExternalUsersGroup =
+          !!allTenantUsersGroup && (isAllTenantUsersGroup || isTenantGroup);
```

Comment:
`!!allTenantUsersGroup` is needed for type check reasons, what really fixes it is `|| isTenantGroup